### PR TITLE
Add tests for 1.6.3.

### PR DESCRIPTION
**Pull request by isuruuy429**
## Purpose
Add tests and relevant artifacts to 1.6.3.

### CODE REVIEW

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/01-SynapseConfigProject/01-synapseConfig/src/main/synapse-config/proxy-services/1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<proxy name="1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty" startOnLoad="true" transports="http https" xmlns="http://ws.apache.org/ns/synapse">
+    <target>
+        <inSequence>
+            <log level="full"/>
```
Comment (sdkottegoda) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/01-SynapseConfigProject/01-synapseConfig/src/main/synapse-config/proxy-services/1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty.xml]: LEt's not have full logs

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/01-SynapseConfigProject/01-synapseConfig/src/main/synapse-config/proxy-services/1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            <property expression="//sam:placeOrder/sam:order" name="testProperty" scope="default" type="OM" xmlns:sam="http://sample.wso2.org"/>
+            <enrich>
+                <source clone="false" property="testProperty" type="property"/>
+                <target type="body"/>
+            </enrich>
+            <log level="full"/>
```
Comment (sdkottegoda) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/01-SynapseConfigProject/01-synapseConfig/src/main/synapse-config/proxy-services/1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty.xml]: Shall we remove all unwanted log fulls?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/pom.xml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                        <phase>compile</phase>
+                        <goals>
+                            <goal>copy-dependencies</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${basedir}/src/test/resources/artifacts/ESB/jar
+                            </outputDirectory>
+                            <includeTypes>jar</includeTypes>
+                            <includeArtifactIds>
+                                groovy-all,jruby-complete,mysql-connector-java,activemq-core,geronimo-j2ee-management_1.1_spec,geronimo-jms_1.1_spec
```
Comment (sdkottegoda) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/pom.xml]: Do we actually need these?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+import java.io.IOException;
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * This class is to test if xml payload can be enriched before it goes to the backend server by replacing elements
+ * in the payload. This can be done using enrich mediator. This class focusses on various ways of achieving that.
+ */
+public class ReplaceElementsTest extends ScenarioTestBase {
+
+    @BeforeClass(alwaysRun = true)
```
Comment (sdkottegoda) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java]: we don't need alwaysRun = true

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    public void init() throws Exception {
+        super.init();
+    }
+
+    //This test is to verify if payload can be modified by replacing the body of payload using a payload stored in a property
+    @Test(description = "1.6.3.1 - Replacing body of payload using a payload stored in a property")
+    public void ReplaceMessageBodyUsingPayloadStoredInProperty() throws IOException, XMLStreamException {
+        String url = getProxyServiceURLHttp("1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty");
+
+        String request =
```
Comment (sdkottegoda) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java]: We just need a simple soap request right? Can we have a string constant in ScenarioConstants and use it wherever possible?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+                        + "   </soapenv:Body>\n"
+                        + "</soapenv:Envelope>";
+
+        HTTPUtils.invokeSoapActionAndAssert(url, request, ScenarioConstants.MESSAGE_ID, expectedResponse, 200,
+                "urn:mediate", "ReplaceTargetDefinedThroughXpathBySourceInlineContent");
+    }
+
+
+    @AfterClass(description = "Server Cleanup",
+                alwaysRun = true)
```
Comment (sdkottegoda) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java]: Let's remove "alwaysRun = true"

Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java]: Done

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java (hunk 5 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java]: Done.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java (hunk 4 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java]: Removed

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/pom.xml (hunk 3 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/pom.xml]: Removed.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/01-SynapseConfigProject/01-synapseConfig/src/main/synapse-config/proxy-services/1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty.xml (hunk 2 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/01-SynapseConfigProject/01-synapseConfig/src/main/synapse-config/proxy-services/1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty.xml]: Removed

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/01-SynapseConfigProject/01-synapseConfig/src/main/synapse-config/proxy-services/1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty.xml (hunk 1 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/01-SynapseConfigProject/01-synapseConfig/src/main/synapse-config/proxy-services/1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty.xml]: Removed.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+                        + "      <sam:order>\n"
+                        + "         <xsd:price>12</xsd:price>\n"
+                        + "         <xsd:productid>IC002</xsd:productid>\n"
+                        + "         <xsd:quantity>2</xsd:quantity>\n"
+                        + "         <xsd:reference>ref</xsd:reference>\n"
+                        + "      </sam:order>\n"
+                        + "   </soapenv:Body>\n"
+                        + "</soapenv:Envelope>";
+
+        HTTPUtils.invokeSoapActionAndAssert(url, request, ScenarioConstants.MESSAGE_ID, expectedResponse, 200,
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java]: Let's add test case ID for message ID header

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioConstants.java [hunk 8]
```diff
@@ -37,6 +37,19 @@
     public static final String VERSION_640 = "EI-6.4.0";
     public static final String VERSION_650_SNAPSHOT = "EI-6.5.0-SNAPSHOT";
     public static final String BASIC_JSON_MESSAGE = "{ \"name\":\"John\", \"age\":30, \"car\":null}";
+    public static final String SOAP_REUEST_1_6_3 = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
```
Comment (milindaperera) [inline product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioConstants.java]: We should not add scenario specific constants here. If this is common placeOrder sample request for SimpleStockQuote service, then fine. But we should change the name to a common name 
Suggested name: SIMPLE_QUOTE_PLACEORDER

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java (hunk 7 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java]: This is done.

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioConstants.java (hunk 8 above)
Comment (isuruuy429) [reply product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioConstants.java]: This is done.

**Comment by isuruuy429** [comment]
Covers from https://github.com/wso2/product-ei/pull/3103

**Pull request closed without merging**